### PR TITLE
Please enable connecting to etcd with tls based authentication

### LINKED ISSUE #13: Error with secured etcd
Hi,
after having successfully tested etcd, patroni and vip-manager on a 3 nodes cluster, I got some issue with vip-manager when etcd is secured.

etcd error: client: etcd cluster is unavailable or misconfigured; error #0: x509: certificate signed by unknown authority

Patroni is configured to use the certificates and is working well.

Could you tell me if vip-manager need some additional configuration to support certificates with etcd ?

And thank you for the tool, very usefull.

Regards,
Henri Chapelle

### PR DESCRIPTION
**Pull request by tpo**
This branch extracts work done by https://github.com/gandalfmagic/vip-manager to allow vip-manager to connect to etcd via tls based authentication.

It should fix #13

### CODE REVIEW

**Comment by pashagolub** [comment]
@tpo  @gandalfmagic Could you please rebase your PR on the latest version, please?

File: vipconfig/config.go [hunk 1]
```diff
@@ -14,10 +14,13 @@ type Config struct {
 	Endpoints    []string `yaml:"endpoints"`
 	EtcdUser     string   `yaml:"etcd_user"`
 	EtcdPassword string   `yaml:"etcd_password"`
+	EtcdCAFile   string   `yaml:etcd_ca_file`
+	EtcdCertFile string   `yaml:etcd_cert_file`
+	EtcdKeyFile  string   `yaml:etcd_key_file`
```
Comment (pashagolub) [inline vipconfig/config.go:19]:
```suggestion
	EtcdCAFile   string   `yaml:"etcd_ca_file"`
	EtcdCertFile string   `yaml:"etcd_cert_file"`
	EtcdKeyFile  string   `yaml:"etcd_key_file"`  
```

File: main.go [hunk 2]
```diff
@@ -64,8 +67,8 @@ func getNetIface(iface string) *net.Interface {
 func main() {
 	flag.Parse()
 
-	if *versionHint {
-		fmt.Println("version 0.6.1")
+	if *versionHint == true {
```
Comment (pashagolub) [inline main.go:70]:
```suggestion
	if *versionHint {
```

**Comment by markwort** [comment]
Merged in 86db6ec88a31f0732bd7e77b1c155f94be6e4fde .
Thanks everyone!

**Pull request closed without merging**